### PR TITLE
Adds parent property of repo endpoint

### PR DESCRIPTION
**Pull request by guyzmo**
cf gogits/gogs#4484
cf gogits/go-gogs-client#64
cf go-gitea/gitea#1687
cf go-gitea/go-sdk#59

### CODE REVIEW

**Comment by guyzmo** [comment]
I just changed the content, to expose Repository* BaseRepo as 'parent' property, instead of an id. because that's more restful.

File: gogs_client/entities.py [hunk 1]
```diff
@@ -172,6 +174,15 @@ def fork(self):
         return self._fork
 
     @property
+    def parent(self):
+        """
+        Gets the repository's parent, when a fork
+
+        :rtype: int
```
Comment (ethantkoenig) [inline gogs_client/entities.py]: No longer an `int`, right?

Comment (guyzmo) [inline gogs_client/entities.py]: yup, solved. Also made it automatically convert to `GogsRepo` if not `None`.

**Comment by ethantkoenig** [comment]
LGTM (coverage decrease is negligible)

**Pull request merged**